mmengine/dist/dist.py

```diff
@@ -1010,8 +1010,10 @@ def collect_results_cpu(result_part: list,
                 part_list.append(pickle.load(f))
         # sort the results
         ordered_results = []
-        for res in zip(*part_list):
-            ordered_results.extend(list(res))
+        zipped_results = zip_longest(*part_list)
+        ordered_results = [
+            i for i in chain.from_iterable(zipped_results) if i is not None
```

Comment:
Why do we need to check whether `is` is not None?

Reply:
`None` is used for padding
```python
from itertools import zip_longest, chain

part_list = [[1, 2], [3, 4, 5, 6, 7], [8, 9, 10]]
zipped_results = zip_longest(*part_list)
ordered_results = [i for i in chain.from_iterable(zipped_results)]
ordered_results
# [1, 3, 8, 2, 4, 9, None, 5, 10, None, 6, None, None, 7, None]
```